packages/cli-kit/src/public/node/session-prompt.ts

```diff
@@ -0,0 +1,95 @@
+import {renderSelectPrompt} from './ui.js'
+import {ensureAuthenticatedUser} from './session.js'
+import {identityFqdn} from './context/fqdn.js'
+import {setCurrentSessionId} from '../../private/node/conf-store.js'
+import * as sessionStore from '../../private/node/session/store.js'
+import type {Sessions} from '../../private/node/session/schema.js'
+
+const NEW_LOGIN_VALUE = 'NEW_LOGIN'
+
+interface SessionChoice {
+  label: string
+  value: string
+}
+
+/**
+ * Builds the choices array from existing sessions.
+ *
+ * @param sessions - The sessions object from storage.
+ * @param fqdn - The identity provider FQDN.
+ * @returns Array of session choices.
+ */
+function buildSessionChoices(sessions: Sessions, fqdn: string): SessionChoice[] {
+  const choices: SessionChoice[] = []
+  const fqdnSessions = sessions[fqdn]
+
+  if (fqdnSessions) {
+    for (const [userId, session] of Object.entries(fqdnSessions)) {
+      choices.push({
+        label: session.identity.alias ?? userId,
+        value: userId,
+      })
+    }
+  }
+
+  return choices
+}
+
+/**
+ * Prompts the user to select from existing sessions or log in with a new account.
+ *
+ * This function:
+ * 1. Fetches existing sessions from storage using `store.fetch()`
+ * 2. Shows a prompt with all available sessions by their display labels
+ * 3. Includes an option to "Log in with a new account"
+ * 4. If an existing session is chosen, calls `setCurrentSessionId(userId)`
+ * 5. If new login is chosen, calls `ensureAuthenticatedUser()`.
+ *
+ * @param alias - Optional alias to use for the new session if created.
+ * @example
+ * ```typescript
+ * import {promptSessionSelect} from '@shopify/cli-kit/node/session-prompt'
+ *
+ * const result = await promptSessionSelect()
+ * console.log(`Using session for user: ${result.userId}`)
+ * ```
+ *
+ * @returns Promise that resolves with the user ID of the selected or newly created session.
+ */
+export async function promptSessionSelect(alias?: string): Promise<{userId: string}> {
+  const sessions = await sessionStore.fetch()
+  const fqdn = await identityFqdn()
+
+  const choices: SessionChoice[] = []
+
+  // Add existing sessions if any
+  if (sessions) {
+    const sessionChoices = buildSessionChoices(sessions, fqdn)
+    choices.push(...sessionChoices)
+  }
+
+  let selectedValue = NEW_LOGIN_VALUE
+
+  if (choices.length > 0) {
+    choices.push({
+      label: 'Log in with a new account',
```

Comment:
```suggestion
      label: 'Log in with a different account',
```
New made me think I was going to kick off a flow to create a brand new user name and password. Can we change it to `different` wherever we use this label?